flintrock/ec2.py

```diff
@@ -647,14 +659,17 @@ def _create_instances(
         subnet_id,
         instance_profile_arn,
         ebs_optimized,
-        instance_initiated_shutdown_behavior) -> 'List[boto3.resources.factory.ec2.Instance]':
+        instance_initiated_shutdown_behavior,
+        user_data) -> 'List[boto3.resources.factory.ec2.Instance]':
     ec2 = boto3.resource(service_name='ec2', region_name=region)
 
     cluster_instances = []
     spot_requests = []
+    user_data = user_data.read()
```

Comment:
To follow-on to my comment on the main PR, we should remove this line and have the methods calling `_create_instances()` take care of reading the file contents where appropriate.